File: lab-1/domain/solo_artist.py
from artist import Artist


class SoloArtist(Artist):
    def __init__(self, name, debut_date, dob, genre, country):
        super().__init__(name, debut_date)
        self.dob = dob
        self.genre = genre
        self.country = country
        self.albums = []

    def introduce(self):
        print(f"Hello, I am {self.name}, a solo artist from {self.country}.")
        print(f"I have been activating in the music industry since {self.debut_date}.")

File: src/main/kotlin/performer/implementations/artist/implementations/SoloArtist.kt
package performer.implementations.artist.implementations
import media_item.implementations.Album
import performer.implementations.artist.Artist

class SoloArtist(
        name: String,
        debutDate: String,
        val dob: String,
        val genre: String,
        val country: String
) : Artist(name, debutDate) {
    val albums = mutableListOf<Album>()

    override fun introduce() {
        println("Hello, I am $name, a solo artist from $country.")
        println("I have been activating in the music industry since $debutDate.")
    }

    override fun toString(): String {
        return "SoloArtist(name='$name', debutDate='$debutDate', dob='$dob', genre='$genre', country='$country')"
    }
}